### ISSUE TITLE
Notifications

### ISSUE THREAD

**Issue by Zefau** [Zefau/ioBroker.jarvis#56]
Add Notifications within jarvis. 

Any message may be added to `jarvis.0.addNotification` is pushed to jarvis and showed to the user.
Depending on a display attribute, the notification may either 
- [x] only be visible within the sidebar drawer
- [x] shown via a Snackbar
- [x] shown via a Popup Dialog

Additionally

- [x] Snackbar or Popup Dialog may be hidden after a defined timeout period.
- [x] The notification may be pushed to the Android / iOS notification drawer (via https://serviceworke.rs/push-subscription-management_service-worker_doc.html).
- [x] Notifications may be archived and deleted via drawer.

Message structure:
``` 
{
  "title":"A title for the notification", // required
  "message":"A message for the notification", // optional

  "state": "unread | read | archived", // optional, one of unread, read or archived (default is unread)
  "criticality":"A criticality, which results in different icons", // optional, one of none, error, warning, info or success (default is none)
  "display":"Either 'drawer', 'snackbar' or 'popup' as described above", // optional, one of drawer, snackbar, popup or none (default is popup)

  "autoHideDuration":"Time in milliseconds the message is shown before disappearing", // in milliseconds
  "snackbarOrigin": { "horizontal": "...", "vertical": "" }, // optional, applies when display is snackbar, { horizontal: 'center' | 'left' | 'right', vertical: 'bottom' | 'top' }

  "focusTab": "tab ID to focus",

  "icon":"Custom icons, which overrides criticality",
  "iconColor": "darkred",
  "backgroundColor": "",
  "fontColor": "blue",

  "focusTab": "focus a certain tab" // tab to focus
  "callback": () => console.log("hello world") // invoke callback
}
```

**Comment by braindead1** [comment]
The idea is awesome!

**Comment by Zefau** [comment]
You may vote it up :)

**Comment by Pix---** [comment]
Great if popups may contain widgets. :-D

**Comment by sicadrip** [comment]
Great idea! Will there be a datapoint for unread messages? I.e. to trigger scripts when a notification has been read/closed.

**Comment by Zefau** [comment]
@sicadrip it will be a state holding a json structure of all notifications with a key indicating if a notification is archived, unread, etc.

**Comment by simatec** [comment]
Is this feature already included in version 2.1.0? I don't get any notifications

**Comment by Zefau** [comment]
@simatec No, it is tagged for v2.2.0 as you can see in the milestone. Why do you think 2.1.0 ?

**Comment by simatec** [comment]
then I got it wrong.  thought it is already integrated because the object already exists

**Comment by Zefau** [comment]
initial release with `v2.2.0-beta.25`

**Comment by Zefau** [comment]
- [ ] change tab / view

**Comment by Zefau** [comment]
https://forum.iobroker.net/post/566747
> Wird eine Benachrichtigung hinzugefügt und es kein Browser mit JARVIS an, wird diese Benachrichtigung nicht in JARVIS angezeigt.

> Popup ist etwas breit: (Vielleicht abhängig vom Inhalt aber max 80% Breite bei hoher Auflösung?)

**Comment by Zefau** [comment]
https://forum.iobroker.net/post/569585

**Comment by Zefau** [comment]
https://forum.iobroker.net/post/570257

**Comment by Zefau** [comment]
finalized initial implementation with `v2.2.0-beta.29`. More to come with https://github.com/Zefau/ioBroker.jarvis/issues/639.

**Comment by Zefau** [comment]
https://forum.iobroker.net/post/571715

**Comment by Zefau** [comment]
- [ ] https://forum.iobroker.net/post/572544

**Comment by Zefau** [comment]
bugs resolved with `2.2.0-beta.33`.

**Comment by Zefau** [comment]
https://forum.iobroker.net/post/587110
